### PR TITLE
Remove exclusivity support for Swift 3 mode.

### PR DESCRIPTION
**Pull request by atrick**
Remove the compiler support for exclusivity warnings.

Leave runtime support for exclusivity warnings in non-release builds
only for unit testing convenience.

Remove a test case that checked the warning log output.

Modify test cases that relied on successful compilation in the
presence of exclusivity violations.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci test.

**Comment by atrick** [comment]
@swift-ci test source compatibility.

**Comment by atrick** [comment]
Fixes <rdar://problem/45146046> Remaining -swift-version 3 tests for exclusivity.

**Comment by atrick** [comment]
@swift-ci test.

**Comment by atrick** [comment]
@swift-ci test source compatibility.

**Comment by slavapestov** [comment]
Thanks!

**Comment by atrick** [comment]
@swift-ci test.

**Comment by atrick** [comment]
@swift-ci test source compatibility.

File: include/swift/ABI/MetadataValues.h [hunk 1]
```diff
@@ -997,8 +997,10 @@ enum class ExclusivityFlags : uintptr_t {
   // Read or Modify).
   ActionMask       = 0x1,
 
-  // Downgrade exclusivity failures to a warning.
+#ifndef NDEBUG
```
Comment (devincoughlin) [inline include/swift/ABI/MetadataValues.h]: One possibility to remove the need to expose this in the ABI header is to separate out the unit tests that abort into a separate testing tool (similar to swift-ide-test) and turn the unit tests into file check tests.

**Review by devincoughlin — APPROVED** [review]
Looks good to me! Some comments in line.

**Comment by atrick** [comment]
@devincoughlin @rjmccall I removed the remaining runtime support for warnings and moved the runtime unit tests into a library.

**Comment by atrick** [comment]
@swift-ci test.

**Comment by atrick** [comment]
@swift-ci test linux platform.

**Comment by atrick** [comment]
@swift-ci test.

**Pull request merged**